code/modules/jobs/job_types/silicon.dm

```diff
@@ -16,8 +16,26 @@ AI
 	exp_type = EXP_TYPE_CREW
 	antag_rep = 20
 
-/datum/job/ai/equip(mob/living/carbon/human/H)
-	return H.AIize(FALSE)
+/datum/job/ai/equip(mob/living/carbon/human/H, latejoin)
+	. = H.AIize(FALSE)
+	if(latejoin)
+		var/obj/structure/AIcore/latejoin_inactive/available
+		for(var/obj/structure/AIcore/latejoin_inactive/P in GLOB.latejoin_ai_cores)
+			if(P.is_available())
+				available = P
+				GLOB.latejoin_ai_cores -= P
+				break
+		if(available)
+			available.available = FALSE
+			var/mob/living/silicon/ai/AI = .
+			AI.forceMove(available.loc)
+			qdel(available)
+		else
+			log_game("User ckey [H.ckey] has been forcefully made into an AI on latejoining as AI, without being put into a latejoin shell. This is either due to admin intervention or a bug/race condition!")
+			message_admins("User ckey [H.ckey] has been forcefully made into an AI on latejoining as AI, without being put into a latejoin shell. This is either due to admin intervention or a bug/race condition!")

```

Comment:
There shouldn't be any race conditions reee

Reply:
but cyber what if there is?!!